netboot/netboot.go

```diff
@@ -46,6 +47,37 @@ func RequestNetbootv6(ifname string, timeout time.Duration, retries int, modifie
 	return conversation, nil
 }
 
+// RequestNetbootv4 sends a netboot request via DHCPv4 and returns the exchanged packets. Additional modifiers
+// can be passed to manipulate both the discover and offer packets.
+func RequestNetbootv4(ifname string, timeout time.Duration, retries int, modifiers ...dhcpv4.Modifier) ([]*dhcpv4.DHCPv4, error) {
+	var (
+		conversation []*dhcpv4.DHCPv4
+		err          error
+	)
+	delay := 2 * time.Second
+	for i := 0; i <= retries; i++ {
+		log.Printf("sending request, attempt #%d", i+1)
+		modifiers = append(modifiers, dhcpv4.WithNetboot)
```

Comment:
nit: I would move this outside of the for to avoid appending the same modifier in the retries.

Apart from this, it looks good to me.